controllers/grid/users/stageParticipant/form/AddParticipantForm.inc.php

```diff
@@ -47,10 +70,34 @@ function getSubmission() {
 		return $this->_submission;
 	}
 
+	/**
+	 * Initialize private attributes that need to be used through all functions.
+	 */
+	function initialize() {
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+
+		// assign all user group IDs with ROLE_ID_MANAGER or ROLE_ID_SUB_EDITOR
+		$this->_managerGroupIds = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_MANAGER, $this->_contextId);
+		$subEditorGroupIds = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_SUB_EDITOR, $this->_contextId);
+		$this->_possibleRecommendOnlyUserGroupIds = array_merge($this->_managerGroupIds, $subEditorGroupIds);
+
+		if ($this->_assignmentId) {
+			/** @var $stageAssignmentDao StageAssignmentDAO */
+			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+
+			/** @var $stageAssignment StageAssignment */
+			$stageAssignment = $stageAssignmentDao->getById($this->_assignmentId);
+			$this->_isChangePermitMetadataAllowed = !in_array($stageAssignment->getUserGroupId(), $this->_managerGroupIds);
+			$this->_isChangeRecommentOnlyAllowed = in_array($stageAssignment->getUserGroupId(), $this->_possibleRecommendOnlyUserGroupIds);
+		}
+	}
+
 	/**
 	 * @copydoc Form::fetch()
 	 */
 	function fetch($request, $template = null, $display = false) {
+		$this->initialize($request);
```

Comment:
This is already called in the constructor; is this second call a mistake?

Reply:
It seems that it's not needed here........